sys/include/net/netopt.h

```diff
@@ -574,6 +574,84 @@ typedef enum {
      */
     NETOPT_PHY_BUSY,
 
+    /**
+     * @brief   (uint8_t*) LoRaWAN device EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_DEVEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN device address (4 bytes length)
+     */
+    NETOPT_LORAWAN_DEVADDR,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_APPEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPKEY,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN network session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_NWKSKEY,
+    /**
+     * @brief   (uint8_t*) LoRaWAN application session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPSKEY,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN join state
+     */
+    NETOPT_LORAWAN_JOIN_STATE,
+
+     /**
+     * @brief   (uint8_t) LoRaWAN device class (A, B, C)
+     * - LoRaWAN: @ref loramac_class_t
+     */
+    NETOPT_LORAWAN_DEVICE_CLASS,
+
+    /**
+     * @brief   (uint8_t) LoRaWAN datarate
+     * - LoRaWAN: @ref loramac_dr_idx_t
+     */
+    NETOPT_LORAWAN_DATARATE,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN adaptive datarate
+     */
+    NETOPT_LORAWAN_ADAPTIVE_DATARATE,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN public network
+     */
+    NETOPT_PUBLIC_NETWORK,
+
+    /**
+     * @brief   (uint32_t) LoRaWAN network id
+     */
+    NETOPT_LORAWAN_NETID,
```

Comment:
I think you forgot one https://github.com/RIOT-OS/RIOT/pull/9022#discussion_r184078312 (I think this can map to `NETOPT_NID`).

Reply:
This is a duplicate do `NETOPT_NID` now

Reply:
indeed. I missed this one